docs/user-guide/src/capm3/node_reuse.md

```diff
@@ -0,0 +1,45 @@
+# Node Reuse
+
+This feature brings a possibility of re-using the same BaremetalHosts (referred to as a host later)
+during deprovisioning/provisioning mainly as part of the rolling upgrade process in the cluster.
+
+## Workflow
+
+Metal3MachineTemplate (M3MT) CRD is the object to support enabling/disabling node reuse feature. When set to **true**,
+Cluster API Provider Metal3 (CAPM3) Machine controller performs the following:
+
+During deprovisioning:
+
+1. Finds M3MT referenced by Metal3Machine (M3M) and checks if nodeReuse is enabled;
+1. If enabled, it checks Cluster API (CAPI) machine object role (control-plane or worker), and based on that, fetches
+   either KubeadmControlPlane (KCP) or MachineDeployment (MD) name accordingly;
+1. Sets fetched object's name on the corresponding host's labels field.
+
+On the following provisioning:
+
+1. Filters out all hosts with node reuse label set and those that store KCP/MD object name, in a format of i.e:
+  **<infrastructure.cluster.x-k8s.io/node-reuse: kcp-object-name>** where after node reuse label, a prefix is
+  appended depending on the owner of the host in question (kcp or md) to avoid mismatch, in cases where KCP and
+  MD object names are the same.
+1. Chooses a host from the filtered list of reused hosts for provisioning.
+
```

Comment:
Can we project the steps on an example ?

Reply:
I am not sure if I got the idea, can you elaborate more?

Reply:
I was thinking if an example that applies those steps on a list of bmh would simplify understanding the node reuse workflow

Reply:
@mboukhalfa having a controller logic re-explained on this document felt over-burden, so I have completely removed the controller logic from the user guide and provided a reference to it, thus I have removed the points we are discussing here